### PR TITLE
feat(general): Implement authentication retry mechanism 

### PR DESCRIPTION
**Pull request by OfekShimko**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
POST methods and 401 are unhandled by default using urllib3. In this MR we include them.

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/bridgecrew/platform_integration.py [hunk 1]
```diff
@@ -55,7 +55,7 @@
     normalize_bc_url,
     REQUEST_CONNECT_TIMEOUT,
     REQUEST_READ_TIMEOUT,
-    REQUEST_RETRIES,
+    REQUEST_RETRIES, REQUEST_STATUS_CODES_RETRY, REQUEST_METHODS_TO_RETRY,
```
Comment (Eliran-Turgeman) [inline checkov/common/bridgecrew/platform_integration.py:58]: are you using these imports? I see youv'e defined them as consts below

File: checkov/common/bridgecrew/platform_integration.py [hunk 2]
```diff
@@ -127,7 +129,12 @@ def clean(self) -> None:
         self.platform_integration_configured = False
         self.http: urllib3.PoolManager | urllib3.ProxyManager | None = None
         self.http_timeout = urllib3.Timeout(connect=REQUEST_CONNECT_TIMEOUT, read=REQUEST_READ_TIMEOUT)
-        self.http_retry = urllib3.Retry(REQUEST_RETRIES, redirect=3)
+        self.http_retry = urllib3.Retry(
```
Comment (Eliran-Turgeman) [inline checkov/common/bridgecrew/platform_integration.py:132]: it seems that there was already an auth retry mechanism and we narrowed the conditions for it, is that intended? (I mean we still do `REQUEST_RETRIES` retries at most which was the case before)

File: checkov/common/bridgecrew/platform_integration.py (hunk 1 above)
Comment (OfekShimko) [reply checkov/common/bridgecrew/platform_integration.py:58]: Right, I'll delete them. 
Thank you!

File: checkov/common/bridgecrew/platform_integration.py (hunk 2 above)
Comment (OfekShimko) [reply checkov/common/bridgecrew/platform_integration.py:132]: POST methods and 401 are unhandled by default using urllib3. In this MR we include them.

**Review by mikeurbanski1 — APPROVED** [review]

**Pull request merged**